ngrok-java/src/main/java/com/ngrok/Session.java

```diff
@@ -33,47 +57,131 @@ public static Session connect(Builder builder) throws IOException {
+     *                creation
+     *                Returns a new {@link TcpTunnel} instance with the specified
+     *                settings
+     * @throws IOException if an I/O error occurs during the tunnel creation
+     */
     public TcpTunnel tcpTunnel(TcpTunnel.Builder builder) throws IOException;
 
+    /**
+     * Creates a new {@link TlsTunnel} instance with the default settings.
+     *
+     * @returns a new {@link TlsTunnel} instance with the default settings
+     * @throws IOException if an I/O error occurs during the tunnel creation
+     */
     public default TlsTunnel tlsTunnel() throws IOException {
         return tlsTunnel(new TlsTunnel.Builder());
     }
 
+    /**
+     * Creates a new {@link TlsTunnel} instance with the specified settings.
+     *
+     * @param builder the {@link TlsTunnel.Builder} instance to use for the tunnel
+     *                creation
+     * @returns a new {@link TlsTunnel} instance with the specified settings
+     * @throws IOException if an I/O error occurs during the tunnel creation
+     */
     public TlsTunnel tlsTunnel(TlsTunnel.Builder builder) throws IOException;
 
+    /**
+     * Creates a new {@link HttpTunnel} instance with the default settings.
+     *
+     * @return a new {@link HttpTunnel} instance with the default settings
+     * @throws IOException if an I/O error occurs during the tunnel creation
+     */
     public default HttpTunnel httpTunnel() throws IOException {
         return httpTunnel(new HttpTunnel.Builder());
     }
-    
+
+    /**
+     * Creates a new {@link HttpTunnel} instance with the specified settings.
+     *
+     * @param builder the {@link HttpTunnel.Builder} instance to use for the tunnel
+     *                creation
+     * @return a new {@link HttpTunnel} instance with the specified settings
+     * @throws IOException if an I/O error occurs during the tunnel creation
+     */
     public HttpTunnel httpTunnel(HttpTunnel.Builder builder) throws IOException;
 
+    /**
+     * Returns a new {@link LabeledTunnel} instance with the default settings.
+     *
+     * @return a new {@link LabeledTunnel} instance with the default settings
+     * @throws IOException if an I/O error occurs during the tunnel creation
+     */
     public default LabeledTunnel labeledTunnel() throws IOException {
         return labeledTunnel(new LabeledTunnel.Builder());
     }
 
+    /**
+     * Returns a new {@link LabeledTunnel} instance with the specified settings.
+     *
+     * @param builder the {@link LabeledTunnel.Builder} instance to use for the tunnel
+     *                creation
+     * @return a new {@link LabeledTunnel} instance with the specified settings
+     * @throws IOException if an I/O error occurs during the tunnel creation
+     */
     public LabeledTunnel labeledTunnel(LabeledTunnel.Builder builder) throws IOException;
 
+    /**
+     * Represents a callback for stopping a session. Implements 
```

Comment:
> A callback used when the agent receives stop session from the ngrok server.   

or maybe just copy the first paragraph from: https://docs.rs/ngrok/latest/ngrok/session/struct.SessionBuilder.html#method.handle_stop_command ? (also for the rest callbacks)